kiwi/builder/container.py

```diff
@@ -136,7 +136,9 @@ def create(self):
             key='container',
             filename=self.filename,
             use_for_bundle=True,
-            compress=False,
+            compress=self.runtime_config.get_bundle_compression(
```

Comment:
Doesn't this mean with force-enabled bundle compression it would result in `.tar.xz.xz` files?

Reply:
I guess it would happen if one enables the compression of the bundle and also the compression of the image. KIWI build can produce compressed or uncompressed container images, thus if one applies compressed images plus compressed bundle, the result will be a double compression.

Reply:
That doesn't make sense, so should IMO not be possible.

Reply:
No because we create container tar's always uncompressed. But the container compression is an option as well. So if you set:

```yaml
container:
      - compress: xz

bundle:
    - compress: true
```

you will end up in the situation you described above. I think the above setup doesn't make much sense but it can be configured that way. Usually I don't prevent people from shooting themselves into their foot ;)

Reply:
Understandable, but apparently the container is compressed by default:

```
[  188s] [ DEBUG   ]: 12:00:01 | EXEC: [skopeo copy oci:/tmp/kiwi_oci_dir.xgvnk2l3/oci_layout:latest docker-archive:/usr/src/packages/KIWI-docker/opensuse-tumbleweed-image.x86_64-1.0.4.docker.tar:opensuse/tumbleweed:latest --additional-tag opensuse/tumbleweed:20190327 --additional-tag opensuse/tumbleweed:20190327.29.13]
[  192s] [ DEBUG   ]: 12:00:05 | EXEC: [xz -f --threads=0 /usr/src/packages/KIWI-docker/opensuse-tumbleweed-image.x86_64-1.0.4.docker.tar]
[  211s] [ INFO    ]: 12:00:24 | Export rpm packages metadata
[  211s] [ DEBUG   ]: 12:00:24 | EXEC: [chroot /usr/src/packages/KIWI-docker/build/image-root rpm -E %_dbpath]
[  211s] [ DEBUG   ]: 12:00:24 | EXEC: [rpm --root /usr/src/packages/KIWI-docker/build/image-root -qa --qf %{NAME}|%{EPOCH}|%{VERSION}|%{RELEASE}|%{ARCH}|%{DISTURL}\n --dbpath /usr/lib/sysimage/rpm]
[  211s] [ INFO    ]: 12:00:24 | Export rpm verification metadata
[  211s] [ DEBUG   ]: 12:00:24 | EXEC: [chroot /usr/src/packages/KIWI-docker/build/image-root rpm -E %_dbpath]
[  211s] [ DEBUG   ]: 12:00:24 | EXEC: [rpm --root /usr/src/packages/KIWI-docker/build/image-root -Va --dbpath /usr/lib/sysimage/rpm]
[  215s] [ DEBUG   ]: 12:00:28 | EXEC: [rm -r -f /tmp/kiwi_oci_root_dir._0cy3ji6]
[  215s] [ INFO    ]: 12:00:28 | Result files:
[  215s] [ INFO    ]: 12:00:28 | --> container: /usr/src/packages/KIWI-docker/opensuse-tumbleweed-image.x86_64-1.0.4.docker.tar.xz
[  215s] [ INFO    ]: 12:00:28 | --> image_packages: /usr/src/packages/KIWI-docker/opensuse-tumbleweed-image.x86_64-1.0.4.packages
[  215s] [ INFO    ]: 12:00:28 | --> image_verified: /usr/src/packages/KIWI-docker/opensuse-tumbleweed-image.x86_64-1.0.4.verified
[  215s] LANG=en_US.UTF-8 /usr/bin/kiwi result bundle --target-dir /usr/src/packages/KIWI-docker --id docker-Build29.13 --bundle-dir //usr/src/packages/KIWI.bundle/ 
[  216s] [ INFO    ]: 12:00:29 | Bundle build results from /usr/src/packages/KIWI-docker
```

No special config is set anywhere.

Reply:
The default container compression is set to compressed, as @Vogtinator noted. Probably we could set container images to be uncompressed and make the bundle compressed by default (the result should be equivalent in terms of OBS) or even drop the compression option for containers images and just have the bundle option.

Reply:
The ability to setup the container compression was requested by iirc b1 systems and we can't just drop that. Having one way to control the container compression is enough and we can simply not evaluate the compress flag set on the bundler for the container builder.